plugins/MLTModule.hpp

```diff
@@ -207,29 +209,50 @@ class MLTModule : public dunedaq::appfwk::DAQModule
   */
   // Opmon variables
   using metric_counter_type = uint64_t ; //decltype(moduleleveltriggerinfo::Info::tc_received_count);
-  std::atomic<metric_counter_type> m_tc_received_count{ 0 };
-  std::atomic<metric_counter_type> m_tc_ignored_count{ 0 };
+  std::atomic<metric_counter_type> m_td_msg_received_count{ 0 };
   std::atomic<metric_counter_type> m_td_sent_count{ 0 };
-  std::atomic<metric_counter_type> m_new_td_sent_count{ 0 };
-  std::atomic<metric_counter_type> m_td_sent_tc_count{ 0 };
+  std::atomic<metric_counter_type> m_td_total_count{ 0 };
+  // DFO state related
   std::atomic<metric_counter_type> m_td_inhibited_count{ 0 };
-  std::atomic<metric_counter_type> m_new_td_inhibited_count{ 0 };
-  std::atomic<metric_counter_type> m_td_inhibited_tc_count{ 0 };
   std::atomic<metric_counter_type> m_td_paused_count{ 0 };
-  std::atomic<metric_counter_type> m_td_paused_tc_count{ 0 };
-  std::atomic<metric_counter_type> m_td_dropped_count{ 0 };
-  std::atomic<metric_counter_type> m_td_dropped_tc_count{ 0 };
-  std::atomic<metric_counter_type> m_td_cleared_count{ 0 };
-  std::atomic<metric_counter_type> m_td_cleared_tc_count{ 0 };
-  std::atomic<metric_counter_type> m_td_not_triggered_count{ 0 };
-  std::atomic<metric_counter_type> m_td_not_triggered_tc_count{ 0 };
-  std::atomic<metric_counter_type> m_td_total_count{ 0 };
-  std::atomic<metric_counter_type> m_new_td_total_count{ 0 };
   std::atomic<metric_counter_type> m_td_queue_timeout_expired_err_count{ 0 };
-  std::atomic<metric_counter_type> m_td_queue_timeout_expired_err_tc_count{ 0 };
+  // livetime related
   std::atomic<metric_counter_type> m_lc_kLive{ 0 };
   std::atomic<metric_counter_type> m_lc_kPaused{ 0 };
   std::atomic<metric_counter_type> m_lc_kDead{ 0 };
+  bool m_lc_started = false;
+
+  // Struct for per TC stats
+  struct TDData {
+    std::atomic<metric_counter_type> received{ 0 };
+    std::atomic<metric_counter_type> sent{ 0 };
+    std::atomic<metric_counter_type> failed_send{ 0 };
+    std::atomic<metric_counter_type> paused{ 0 };
+    std::atomic<metric_counter_type> inhibited{ 0 };
+  };
+  static std::set<trgdataformats::TriggerCandidateData::Type> unpack_types( decltype(dfmessages::TriggerDecision::trigger_type) t) {
```

Comment:
```suggestion
  static std::set<trgdataformats::TriggerCandidateData::Type> unpack_types(const dfmessages::trigger_type_t& t) {
```

Why using `decltype` here at all? We know it's `dfmessages::trigger_type_t`, and it is not going to change. We do use `trigger_type_t` everywhere else for `trigger_type`.

I would only use `decltype` here if e.g. we knew `TriggerDecision::trigger_type` type would keep changing from `dfmessages::trigger_type_t` to completely different types.